File: 1578minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int prev=0;
        int cost=0;
        int colorsL = colors.length();
        for(int i=1;i<colorsL;++i){
            if(colors[i]==colors[prev]){
                int ithCost=neededTime[i];
                int prevthCost=neededTime[prev];
                if(ithCost<prevthCost){
                    cost+=ithCost;
                }else{
                    cost+=prevthCost;
                    prev=i;
                }
            }else{
                prev=i;
            }
        }
        return cost;
    }
};

File: 1578minimum-time-to-make-rope-colorful.cs
public class Solution {
    public int MinCost(string colors, int[] neededTime) {
        int cost =0;
        int prev =0;
        for(int i=1;i<colors.Length;++i){
            if(colors[prev] == colors[i]){
                if(neededTime[i]<neededTime[prev]){
                    cost+=neededTime[i];
                }else{
                    cost+=neededTime[prev];
                    prev=i;
                }
            }else{
                prev=i;
            }
        }
        return cost;
    }
}
